samples/console_output/main.cpp

```diff
@@ -1,38 +1,140 @@
+      error == sensr::MessageListener::Error::kPointResultConnection ) {
+      client_->Reconnect();
+    }
+  }
+  void OnGetOutpuMessage(const sensr_proto::OutputMessage &message) {
+    if (message.has_event()) {
+      for(const auto& zone_event : message.event().zone()) {
+        if (zone_event.type() == sensr_proto::ZoneEvent_Type_ENTRY) {
+          std::cout << "Entering Zone(" << zone_event.id() << ") : obj( " << zone_event.object().id() << ")" << std::endl;
+        } else if (zone_event.type() == sensr_proto::ZoneEvent_Type_EXIT) {
+          std::cout << "Exiting Zone(" << zone_event.id() << ") : obj( " << zone_event.object().id() << ")" << std::endl;
+        }
+      }
+    }
+  }
+private:
+  sensr::Client* client_;
+};
+
+class PointResultListener : public sensr::MessageListener {
+public:
+  PointResultListener(sensr::Client* client) : MessageListener(ListeningType::kPointResult), client_(client) {}
+  void OnError(Error error, const std::string& reason) {
+    if (error == sensr::MessageListener::Error::kOutputMessageConnection || 
+      error == sensr::MessageListener::Error::kPointResultConnection ) {
+      client_->Reconnect();
+    }
+  }
+  void OnGetPointResult(const sensr_proto::PointResult &message) {
+    for(const auto& point_cloud : message.points()) {
+      if (point_cloud.type() == sensr_proto::PointResult_PointCloud_Type_RAW) {
+        std::cout << "Topic(" << point_cloud.id() << ") no. of points - " << point_cloud.points().length() / (sizeof(float) * 3) << std::endl;
+      } else if (point_cloud.type() == sensr_proto::PointResult_PointCloud_Type_GROUND) {
+        std::cout << "Gound points no. of points - " << point_cloud.points().length() / (sizeof(float) * 3) << std::endl;
+      } else if (point_cloud.type() == sensr_proto::PointResult_PointCloud_Type_ENVIRONMENT) {
+        std::cout << "Environment points no. of points - " << point_cloud.points().length() / (sizeof(float) * 3) << std::endl;
+      }
+    }
+  }
+private:
+  sensr::Client* client_;
+};
+
+class ObjectListener : public sensr::MessageListener {
+public:
+  ObjectListener(sensr::Client* client) : MessageListener(ListeningType::kOutputMessage), client_(client) {}
+  void OnError(Error error, const std::string& reason) {
+    if (error == sensr::MessageListener::Error::kOutputMessageConnection || 
+      error == sensr::MessageListener::Error::kPointResultConnection ) {
+      client_->Reconnect();
+    }
+  }
+  void OnGetOutpuMessage(const sensr_proto::OutputMessage &message) {
+    if (message.has_stream()) {
+      for(const auto& object : message.stream().objects()) {
+        int object_points_size = (object.points().length() / (sizeof(float) * 3));
+        std::cout << "Obj(" << object.id() << ") : point no. " << object_points_size << std::endl;
+      }
+    }
+  }
+private:
+  sensr::Client* client_;
+};
+
+#if defined(__linux__)
+#include <sys/time.h>
+#endif
+#include <google/protobuf/util/time_util.h>
```

Comment:
Could you move these headers to the top of the file?

Reply:
Done